validation/src/static/scripts/summary.ts

```diff
@@ -0,0 +1,69 @@
+(async () => {
+  const getElem = (id: string) => document.getElementById(id);
+
+  const urlString = window.location.href;
+  const summaryId = urlString.substr(urlString.lastIndexOf('/') + 1);
+  const rawResponse = await fetch(
+    `http://localhost:3000/api/summary/${summaryId}`
+  );
+  const summary: UrlSummary = await rawResponse.json();
+
+  const urlContainer = getElem('urlContainer');
+  const numNodesContainer = getElem('numNodes');
+  const categoryTable = getElem('categoryTable');
+  if (!urlContainer || !numNodesContainer || !categoryTable)
+    throw new ElemNotFound();
+
+  urlContainer.innerText = summary.url;
+  numNodesContainer.innerText = summary.nodesOnPage.toString();
+
+  const stats = summary.stats.sort(compareStats);
+
+  for (let i = 0; i < stats.length; ++i) {
+    const stat = stats[i];
+    const categoryString = getFormattedCategoryHTML(stat.category);
+
+    // Add a row of generated HTML to the category table.
+    categoryTable.innerHTML += `
+      <tr>
+        <td>${categoryString}</td>
+        <td style="text-align:center;">${stat.count}</td>
+        <td><a href="/case/${stat.caseId}" class="invisibleLink" aria-label="View case ${stat.caseId}"><div class="previewCard" style="margin:0;">Case ${stat.caseId} <span style="margin-left:auto;">🡆</span></div></a></td>
```

Comment:
Optional: Open these in a new tab and use the [open_in_new](https://material.io/resources/icons/?search=new&icon=open_in_new&style=baseline) icon